internal/pkg/client/interface.go

```diff
@@ -49,6 +49,9 @@ type Writer interface {
 	// Patch patches the given obj in the Kubernetes cluster. obj must be a
 	// struct pointer so that obj can be updated with the content returned by the Server.
 	Patch(ctx context.Context, obj runtime.Object, patch Patch, options *metav1.PatchOptions) error
+
+	// Apply uess a merge patch to apply an object
```

Comment:
Should we also mention here (on high level) how `apply` works underneath like 3-way merge?

Reply:
This explanation will be included in the client package.